OpenRA.Game/Primitives/PriorityQueue.cs

```diff
@@ -13,15 +13,31 @@
 
 namespace OpenRA.Primitives
 {
-	public class PriorityQueue<T>
-			where T : IComparable<T>
+	public interface IPriorityQueue<T>
+	{
+		void Add(T item);
+		bool Empty { get; }
+		T Peek();
+		T Pop();
+	}
+
+	public class PriorityQueue<T> : IPriorityQueue<T>
 	{
 		List<T[]> items = new List<T[]>();
 		int level, index;
 
+		private IComparer<T> comparer;
+
 		public PriorityQueue()
 		{
 			items.Add(new T[1]);
+			comparer = Comparer<T>.Default;
```

Comment:
Chain in a way like using the : this() over the parametrized constructor? How can I do that given that both assign different values to the comparer member? If I do that, comparer will be assigned twice. I'm not sure whether that is more elegant...

One way would be to created a shared private function that inits the list, but it's a needlessly complex operation...